### PR TITLE
Mes 8333 adi3 unsuccessful

### PR DESCRIPTION
**Pull request by turnerc1**
## Description

Unsuccessful test with insufficient score 
Unsuccessful test with insufficient risk management
## Checklist

- [ ] PR title includes the JIRA ticket number and an understandable description
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket

### CODE REVIEW

File: src/pageobjects/des/login.pageobject.ts [hunk 1]
```diff
@@ -162,50 +173,29 @@ class LoginMobilePageObject {
...
+    await this.clickNativeButtonWithText('Cancel');
 
-    const nativeCancelButton = await $('//XCUIElementTypeButton[@name="Cancel"]');
-    await this.clickNativeButton(nativeCancelButton);
-    await this.switchToDESContext();
-    const signInAgainButton = await $('span=Sign in');
-    await this.clickElement(signInAgainButton);
+    await browser.pause(3000);
+    // await this.switchToDESContext();
+    console.log('>>>>>>>>>SIGNED OUT>>>>>>>>>');
```
Comment (JCarterDVSA) [inline src/pageobjects/des/login.pageobject.ts:196]: delete commented code

**Review by JCarterDVSA — COMMENTED** [review]
delete commented code

**Review by JCarterDVSA — APPROVED** [review]

File: src/pageobjects/des/testReport.pageobject.ts [hunk 2]
```diff
@@ -330,6 +331,28 @@ class TestReportPageObject extends Page {
     await scroll(selector);
     await clickElement('click', 'selector', selector);
   }
+
+  async keyboardClickLetter(): Promise<void> {
```
Comment (Cullima) [inline src/pageobjects/des/testReport.pageobject.ts:335]: doesn't feel like this method belongs in a page object if it is a generic function for pressing an ios keyboard key.  Can you add a description for the method so people know how to use it and what it does.  Also, worth looking at simpler methods for interacting with the keyboard, e.g. https://webdriver.io/docs/api/appium/#hidekeyboard

**Review by Cullima — CHANGES_REQUESTED** [review]

File: src/pageobjects/des/testReport.pageobject.ts (hunk 2 above)
Comment (turnerc1) [reply src/pageobjects/des/testReport.pageobject.ts]: I have switched out the original function contents to driver.hideKeyboard().

**Review by JCarterDVSA — APPROVED** [review]

File: src/pageobjects/des/login.pageobject.ts [hunk 3]
```diff
@@ -88,6 +88,14 @@ class LoginMobilePageObject {
     await driver.switchContext(DESContext.id);
   }
 
+  async clickNativeButtonWithText(text: string): Promise<void> {
```
Comment (Cullima) [inline src/pageobjects/des/login.pageobject.ts:91]: this seems generic enough to move to a base page object ?

File: src/pageobjects/des/testReport.pageobject.ts [hunk 4]
```diff
@@ -330,6 +331,20 @@ class TestReportPageObject extends Page {
     await scroll(selector);
     await clickElement('click', 'selector', selector);
   }
+
+  /**
+   * Inputs a letter from the IOS keyboard and after closes down the keyboard.
+   * To use this you will have to click on input box element bring up the keyboard.
+   * This function only allows one letter to be typed.
+   */
+  async keyboardClickLetter(letter:string): Promise<void> {
```
Comment (Cullima) [inline src/pageobjects/des/testReport.pageobject.ts:340]: can we also add this to the base page object seems fairly gneric

File: src/pageobjects/des/login.pageobject.ts [hunk 5]
```diff
@@ -1,38 +1,20 @@
...
 import * as credentials from '../../../creds/credentials.json';
+import Page from '../base/page';
 
-export interface Context {
-  id: string,
-  title: string,
-  url: string,
-  bundleId: string,
-}
+// export interface Context {
```
Comment (Cullima) [inline src/pageobjects/des/login.pageobject.ts:6]: can you delete if not needed

**Review by Cullima — APPROVED** [review]

**Review by JCarterDVSA — APPROVED** [review]

**Pull request merged**